src/dataServices/dynamoDbUtil.ts

```diff
@@ -39,6 +42,19 @@ export class DynamoDbUtil {
         }
         item[DOCUMENT_STATUS_FIELD] = documentStatus;
         item[LOCK_END_TS_FIELD] = Date.now();
+
+        // Format of flattenedResource
+        // https://www.npmjs.com/package/flat
+        // flatten({ key1: { keyA: 'valueI' } })  => { key1.keyA: 'valueI'}
+        const flattenedResources: Record<string, string> = flatten(resource);
+        const references = Object.keys(flattenedResources)
+            .filter((key: string) => {
+                return key.includes('reference');

```

Comment:
This is a bug. A field can contain the string "reference" without being a reference i.e. `MolecularSequence.referenceSeq`

https://www.hl7.org/fhir/molecularsequence.html

Reply:
Good catch. I added a fix for it. And added unit test to catch this case.

The fix is to do an exact match on "reference" and to check that "reference" is the last value in the key.